src/set_theory/game.lean

```diff
@@ -0,0 +1,1130 @@
+def le_lt (x y : pgame) : Prop × Prop :=
+begin
+  induction x with xl xr xL xR IHxl IHxr generalizing y,
+  induction y with yl yr yL yR IHyl IHyr,
+  -- the orderings of the clauses here are carefully chosen so that
+  --   and.left/or.inl refer to moves by Left, and
+  --   and.right/or.inr refer to moves by Right.
+  exact ((∀ i, (IHxl i ⟨yl, yr, yL, yR⟩).2) ∧ (∀ j, (IHyr j).2),
+         (∃ i, (IHyl i).1) ∨ (∃ j, (IHxr j ⟨yl, yr, yL, yR⟩).1))
+end
+
+instance : has_le pgame := ⟨λ x y, (le_lt x y).1⟩
+instance : has_lt pgame := ⟨λ x y, (le_lt x y).2⟩
+
+/-- Definition of `x ≤ y` on pre-games built using the constructor. -/
+@[simp] theorem mk_le_mk {xl xr xL xR yl yr yL yR} :
+  (⟨xl, xr, xL, xR⟩ : pgame) ≤ ⟨yl, yr, yL, yR⟩ ↔
+  (∀ i, xL i < ⟨yl, yr, yL, yR⟩) ∧
+  (∀ j, (⟨xl, xr, xL, xR⟩ : pgame) < yR j) := iff.rfl
+
+/-- Definition of `x ≤ y` on pre-games, in terms of `<` -/
+theorem le_def_lt {x y : pgame} : x ≤ y ↔
+  (∀ i : x.left_moves, x.move_left i < y) ∧
+  (∀ j : y.right_moves, x < y.move_right j) :=
+begin
+  induction x with xl xr xL xR IHxl IHxr generalizing y,
+  induction y with yl yr yL yR IHyl IHyr,
+  simp,
+  refl,
+end
+
+/-- Definition of `x < y` on pre-games built using the constructor. -/
+@[simp] theorem mk_lt_mk {xl xr xL xR yl yr yL yR} :
+  (⟨xl, xr, xL, xR⟩ : pgame) < ⟨yl, yr, yL, yR⟩ ↔
+  (∃ i, (⟨xl, xr, xL, xR⟩ : pgame) ≤ yL i) ∨
+  (∃ j, xR j ≤ ⟨yl, yr, yL, yR⟩) := iff.rfl
+
+/-- Definition of `x < y` on pre-games, in terms of `≤` -/
+theorem lt_def_le {x y : pgame} : x < y ↔
+  (∃ i : y.left_moves, x ≤ y.move_left i) ∨
+  (∃ j : x.right_moves, x.move_right j ≤ y) :=
+begin
+  induction x with xl xr xL xR IHxl IHxr generalizing y,
+  induction y with yl yr yL yR IHyl IHyr,
+  simp,
+  refl,
+end
+
+/-- The definition of `x ≤ y` on pre-games, in terms of `≤` two moves later. -/
+theorem le_def {x y : pgame} : x ≤ y ↔
+  (∀ i : x.left_moves,
+   (∃ i' : y.left_moves, (x.move_left i) ≤ (y.move_left i')) ∨
+   (∃ j : (x.move_left i).right_moves, (x.move_left i).move_right j ≤ y)) ∧
+  (∀ j : y.right_moves,
+   (∃ i : (y.move_right j).left_moves, x ≤ (y.move_right j).move_left i) ∨
+   (∃ j' : x.right_moves, x.move_right j' ≤ (y.move_right j))) :=
+begin
+  rw [le_def_lt],
+  conv { to_lhs, simp [lt_def_le] },
+end
+
+/-- The definition of `x < y` on pre-games, in terms of `<` two moves later. -/
+theorem lt_def {x y : pgame} : x < y ↔
+  (∃ (i : left_moves y),
+    (∀ (i' : left_moves x), move_left x i' < move_left y i) ∧
+    (∀ (j : right_moves (move_left y i)), x < move_right (move_left y i) j)) ∨
+  (∃ (j : right_moves x),
+    (∀ (i : left_moves (move_right x j)), move_left (move_right x j) i < y) ∧
+    (∀ (j' : right_moves y), move_right x j < move_right y j')) :=
+begin
+  rw [lt_def_le],
+  conv { to_lhs, simp [le_def_lt] },
+end
+
+-- Possibly the next two lemmas belong elsewhere, but I'm not sure where.
+theorem forall_pempty {P : pempty → Prop} : (∀ x : pempty, P x) ↔ true :=
```

Comment:
`pempty` is defined in logic.basic, and these results could go there I guess.

Reply:
The only problem there is that I can't use `rintro` in that file, because of import dependencies. I could:

1. not use `rintro`
2. try to untangle the import dependencies
3. leave them here
4. find somewhere else

Reply:
Okay, I removed `rintro` and moved them to `logic/basic.lean`.